pkg/controller.v1/pytorch/job.go

```diff
@@ -155,13 +155,16 @@ func (pc *PyTorchController) deletePodsAndServices(job *pyv1.PyTorchJob, pods []
 		return nil
 	}
 
-	// Delete nothing when the cleanPodPolicy is None or Running.
-	if *job.Spec.CleanPodPolicy == common.CleanPodPolicyNone ||
-		*job.Spec.CleanPodPolicy == common.CleanPodPolicyRunning {
+	// Delete nothing when the cleanPodPolicy is None.
+	if *job.Spec.CleanPodPolicy == common.CleanPodPolicyNone {
 		return nil
 	}
 
 	for _, pod := range pods {
+		// Just delete running pod when the cleanPodPolicy is Running
+		if *job.Spec.CleanPodPolicy == common.CleanPodPolicyRunning && pod.Status.Phase != v1.PodRunning {
```

Comment:
Maybe we could add new test case, like this: https://github.com/kubeflow/pytorch-operator/blob/76a240f180da9fcddfcc87bdd8a18b0a4fe33da3/pkg/controller.v1/pytorch/job_test.go#L221-L240 when `*job.Spec.CleanPodPolicy == common.CleanPodPolicyRunning && pod.Status.Phase != v1.PodRunning` ?

Reply:
3q, I've recommitted